app/src/main/kotlin/io/homeassistant/companion/android/webview/WebViewActivity.kt

```diff
@@ -1662,14 +1663,26 @@ class WebViewActivity : BaseActivity(), io.homeassistant.companion.android.webvi
         if (presenter.isAlwaysShowFirstViewOnAppStartEnabled() &&
             LifecycleHandler.isAppInBackground()
         ) {
-            // Pattern matches urls which are NOT allowed to show the first view after app is started
-            // This is
-            // /config/* as these are the settings of HA but NOT /config/dashboard. This is just the overview of the HA settings
-            // /hassio/* as these are the addons section of HA settings.
+            /**
+             * Pattern matches urls which are NOT allowed to show the first view after app is started
+             * This is
+             * /config/ as these are the settings of HA but NOT /config/dashboard. This is just the overview of the HA settings
+             * /hassio/ as these are the addons section of HA settings.
+             */
             if (webView.url?.matches(".*://.*/(config/(?!\\bdashboard\\b)|hassio)/*.*".toRegex()) == false) {
                 Timber.d("Show first view of default dashboard.")
-                webView.evaluateJavascript(
-                    """
+                if (serverManager.getServer(presenter.getActiveServer())?.version?.isAtLeast(2025, 6, 0) == true) {
+                    /**
+                     * Clearing history and replace the current page with the default page from the frontend.
+                     * This way the user have a clear history stack.
+                     */
+                    webView.clearHistory()
```

Comment:
If you're doing this for the new implementation, I'd also change the behavior for the old implementation so it behaves the same.